### PR TITLE
JetMET POG recommended JES

### PR DESCRIPTION
**Pull request by bryates**
This PR includes the JES uncertainties recommended by the POG:
`JES_FlavorPureGluon`, `JES_FlavorPureQuark`, `JES_FlavorPureCharm`, `JES_FlavorPureBottom`, and `JES_TotalNoFlavor`.

### CODE REVIEW

File: analysis/topEFT/topeft.py [hunk 1]
```diff
@@ -275,11 +275,12 @@ def process(self, events):
...
         # Define the lists of systematics we include
         obj_correction_syst_lst = [
-            f'JER_{year}Up',f'JER_{year}Down','JESUp','JESDown' # Systs that affect the kinematics of objects
+            f'JER_{year}Up',f'JER_{year}Down', # Systs that affect the kinematics of objects
+            'FlavorPureGluon', 'FlavorPureQuark', 'FlavorPureCharm', 'FlavorPureBottom', 'TotalNoFlavor'
         ]
         wgt_correction_syst_lst = [
             "lepSF_muonUp","lepSF_muonDown","lepSF_elecUp","lepSF_elecDown",f"btagSFbc_{year}Up",f"btagSFbc_{year}Down","btagSFbc_corrUp","btagSFbc_corrDown",f"btagSFlight_{year}Up",f"btagSFlight_{year}Down","btagSFlight_corrUp","btagSFlight_corrDown","PUUp","PUDown","PreFiringUp","PreFiringDown",f"triggerSF_{year}Up",f"triggerSF_{year}Down", # Exp systs
-            "FSRUp","FSRDown","ISRUp","ISRDown","renormfactUp","renormfactDown", "renormUp","renormDown","factUp","factDown", # Theory systs
+            #"FSRUp","FSRDown","ISRUp","ISRDown","renormfactUp","renormfactDown", "renormUp","renormDown","factUp","factDown", # Theory systs
```
[the remark this answers is not in the document]
Comment (bryates) [reply analysis/topEFT/topeft.py:283]: No, sorry, that was a mistake while testing.

File: topcoffea/modules/corrections.py [hunk 2]
```diff
@@ -495,6 +499,9 @@ def ApplyJetSystematics(year,cleanedJets,syst_var):
   elif(syst_var == 'JESDown'): return cleanedJets.JES_jes.down
   elif(syst_var == 'nominal'): return cleanedJets
   elif(syst_var in ['nominal','MuonESUp','MuonESDown']): return cleanedJets
+  # Save `2016APV` as `2016APV` but look up `2016` corrections (no separate APV corrections available)
+  elif('Up' in syst_var and syst_var.replace('Up', '').replace('APV', '') in cleanedJets.fields): return cleanedJets[syst_var.replace('Up', '').replace('APV', '')].up
+  elif('Down' in syst_var and syst_var.replace('Down', '').replace('APV', '') in cleanedJets.fields): return cleanedJets[syst_var.replace('Down', '').replace('APV', '')].down
```
Comment (andrea21z) [inline topcoffea/modules/corrections.py:504]: The 'APV' replace is not needed

**Comment by andrea21z** [comment]
Thank you @bryates. Unless we want to apply the JES unc. by jet flavor, it looks good to me.

File: topcoffea/modules/corrections.py [hunk 3]
```diff
@@ -498,6 +498,17 @@ def ApplyJetSystematics(year,cleanedJets,syst_var):
   elif(syst_var == 'nominal'): return cleanedJets
   elif(syst_var in ['nominal','MuonESUp','MuonESDown']): return cleanedJets
   # Save `2016APV` as `2016APV` but look up `2016` corrections (no separate APV corrections available)
+  elif('Flavor' in syst_var and 'Up' in syst_var and syst_var.replace('Up', '') in cleanedJets.fields or 'Down' in syst_var and syst_var.replace('Down', '') in cleanedJets.fields):
+      if 'Gluon' in syst_var: mask = abs(cleanedJets.partonFlavour)==21
+      elif 'Bottom' in syst_var: mask = abs(cleanedJets.partonFlavour)==5
+      elif 'Charm' in syst_var: mask = abs(cleanedJets.partonFlavour)==4
+      elif 'Quark' in syst_var: mask = abs(cleanedJets.partonFlavour)<4
```
Comment (kmohrman) [inline topcoffea/modules/corrections.py:505]: Should we have an `else` here? I'm wondering if there is a case where you expect something besides `Gluon`, `Bottom`, `Charm`, or `Quark` in the name? If yes, what would we want to set the mask equal to in that case? If no, then can we raise an exception here (just to make sure nothing unexpected is going on)?

Comment (bryates) [reply topcoffea/modules/corrections.py:505]: No, there's nothing else in the JES files. I can add an exception for bug checking.

**Comment by bryates** [comment]
Closing in favor of #297.

**Pull request closed without merging**